sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/src/SchemaRegistryAvroEventSource.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using Avro;
+using Azure.Core.Diagnostics;
+
+namespace Microsoft.Azure.Data.SchemaRegistry.ApacheAvro
+{
+    /// <summary>
+    ///   Serves as an ETW event source for logging of information about
+    ///   Entity's client.
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   When defining Start/Complete tasks, it is highly recommended that the
+    ///   the CompleteEvent.Id must be exactly StartEvent.Id + 1.
+    /// </remarks>
+    [EventSource(Name = EventSourceName)]
+    internal class SchemaRegistryAvroEventSource : AzureEventSource
+    {
+        /// <summary>The name to use for the event source.</summary>
+        private const string EventSourceName = "Microsoft-Azure-Data-SchemaRegistry-ApacheAvro";
+
+        internal const int CacheUpdatedEvent = 1;
+
+        /// <summary>
+        ///   Provides a singleton instance of the event source for callers to
+        ///   use for logging.
+        /// </summary>
+        public static SchemaRegistryAvroEventSource Log { get; } = new SchemaRegistryAvroEventSource();
+
+        /// <summary>
+        ///   Prevents an instance of the <see cref="SchemaRegistryAvroEventSource"/> class from being
+        ///   created outside the scope of the <see cref="Log" /> instance, as well as setting up the
+        ///   integration with AzureEventSourceListener.
+        /// </summary>
+        protected SchemaRegistryAvroEventSource() : base(EventSourceName)
+        {
+        }
+
+        [NonEvent]
+        public virtual void CacheUpdated(LruCache<string, (Schema Schema, int SchemaLength)> idToSchemaCache, LruCache<Schema, (string SchemaId, int SchemaLength)> schemaToIdCache)
+        {
+            if (IsEnabled())
+            {
+                int totalSchemaLength = idToSchemaCache.Sum(kvp => kvp.Value.SchemaLength);
+                totalSchemaLength += schemaToIdCache.Sum(kvp => kvp.Value.SchemaLength);
+                CacheUpdatedCore(idToSchemaCache.Count + schemaToIdCache.Count, totalSchemaLength);
```

Comment:
Do we want to calculate this every time we log, or could the totals be cached by the caller?

Reply:
We could cache, but this is not a hot path - it is only called when the cache is updated.